### ISSUE TITLE
oce.plot.ts messes up colors when xlim arg is used

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1509]
I discovered this while dealing with #1508 . I'm not sure if this is new since the fix for #1508 , or if it's always been there (would have to revert to check), but I suspect it's the latter.

Because `oce.plot.ts()` actually trims the series when `xlim` is provided, if a `col=` argument is provided where `length(col)=length(x)` then the colors get messed up because they don't get similarly trimmed.

Example and `oce-issues` code coming in next comment.

**Comment by richardsc** [comment]
Example code provided at [oce-issues/1509](https://github.com/dankelley/oce-issues/tree/master/15xx/1509):

```r
library(oce)

t <- seq(as.POSIXct('2019-03-01 00:00:00', tz='UTC'),
         as.POSIXct('2019-03-02 00:00:00', tz='UTC'), 60)
y <- rnorm(t)
z <- seq_along(y)

if (!interactive()) png('1509a-%03d.png')

cm <- colormap(y)
par(mfrow=c(2, 1))
plot(t, y, col=cm$zcol, pch=19, cex=0.5)
title('No trimming -- colours are the same in each plot')
oce.plot.ts(t, y, type='p', col=cm$zcol, pch=19, cex=0.5)

focus <- as.POSIXct(c('2019-03-01 06:00:00', '2019-03-01 18:00:00'), tz='UTC')

plot(t, y, col=cm$zcol, pch=19, cex=0.5, xlim=focus)
title('xlim provided; colours look ok')
oce.plot.ts(t, y, type='p', col=cm$zcol, pch=19, cex=0.5, xlim=focus)
title('xlim provided; colours are mixed up')

if (!interactive()) dev.off()
```
![1509a-002](https://user-images.githubusercontent.com/233584/54484253-60770e00-4841-11e9-9413-b256581682a7.png)

**Comment by richardsc** [comment]
This would be an easy fix, except that `col=` is not a provided argument in `oce.plot.ts()` but rather is captured in the `...`.

To address this I'm going to define it as an explicit arg, and then sort out the case where `xlim` is given and `length(col) == length(x)`.

I'll work in branch 1509, which can be merged if it seems reasonable (and chucked if not).

**Comment by richardsc** [comment]
Fixed in PR #1510, commit af74a006594608. Developers can re-open if they think I'm off base here

**Comment by richardsc** [comment]
![1509a-002](https://user-images.githubusercontent.com/233584/54484939-9242a180-484e-11e9-85a0-a22dfd1cac40.png)

**Comment by richardsc** [comment]
Now, to fix the over-trimming issue from #1508 , specifically [this](https://github.com/dankelley/oce/issues/1508#issuecomment-473612669)
